### PR TITLE
Add type checking to frontend CI & fix `Workflow` relay component

### PR DESCRIPTION
**Pull request by garryod**
Depends on #233 

Typescript CI was previously missing, allowing several typing issues to slip through
Required some fixes to `frontend/tsconfig.json` to have it delegate to package `tsconfig`s
Identified possibly null return on Workflow relay fragment, which is now handled with an `unknown` state

### CODE REVIEW

**Review by daurer — DISMISSED** [review]
LGTM

File: frontend/workflows-lib/tests/components/TasksDynamic.test.tsx [hunk 1]
```diff
@@ -1,31 +1,31 @@
 import { render, screen } from "@testing-library/react";
+import "@testing-library/jest-dom";
```
Comment (daurer) [inline frontend/workflows-lib/tests/components/TasksDynamic.test.tsx:2]: is this import still needed?

File: frontend/workflows-lib/tests/components/TasksFlow.test.tsx [hunk 2]
```diff
@@ -1,30 +1,26 @@
 import { render } from "@testing-library/react";
+import "@testing-library/jest-dom";
```
Comment (daurer) [inline frontend/workflows-lib/tests/components/TasksFlow.test.tsx:2]: still needed?

File: frontend/workflows-lib/tests/components/TasksTable.test.tsx [hunk 3]
```diff
@@ -1,13 +1,9 @@
 import { render } from "@testing-library/react";
+import "@testing-library/jest-dom";
```
Comment (daurer) [inline frontend/workflows-lib/tests/components/TasksTable.test.tsx:2]: same here, still needed?

File: frontend/workflows-lib/tests/components/WorkflowAccordion.test.tsx [hunk 4]
```diff
@@ -1,13 +1,15 @@
 import { render, fireEvent } from "@testing-library/react";
+import "@testing-library/jest-dom";
```
Comment (daurer) [inline frontend/workflows-lib/tests/components/WorkflowAccordion.test.tsx:2]: and here

**Review by daurer — COMMENTED** [review]
Looks good, just wondering if `jest-dom` imports are still needed?

File: frontend/workflows-lib/tests/components/TasksDynamic.test.tsx (hunk 1 above)
Comment (garryod) [reply frontend/workflows-lib/tests/components/TasksDynamic.test.tsx:2]: Yeah, weirdly `jest-dom` isn't actually `jest` specific and is recommended as the to test the dom by `testling-library`, see: https://testing-library.com/docs/svelte-testing-library/setup/#vitest

File: frontend/workflows-lib/tests/components/TasksFlow.test.tsx (hunk 2 above)
Comment (garryod) [reply frontend/workflows-lib/tests/components/TasksFlow.test.tsx:2]: Yeah, weirdly `jest-dom` isn't actually `jest` specific and is recommended as the to test the dom by `testling-library`, see: https://testing-library.com/docs/svelte-testing-library/setup/#vitest

File: frontend/workflows-lib/tests/components/WorkflowAccordion.test.tsx (hunk 4 above)
Comment (garryod) [reply frontend/workflows-lib/tests/components/WorkflowAccordion.test.tsx:2]: Yeah, weirdly `jest-dom` isn't actually `jest` specific and is recommended as the to test the dom by `testling-library`, see: https://testing-library.com/docs/svelte-testing-library/setup/#vitest

**Review by daurer — APPROVED** [review]

**Pull request merged**